fdbcli/fdbcli.actor.cpp

```diff
@@ -2386,29 +2414,54 @@ ACTOR Future<bool> exclude(Database db,
 	} else {
 		state std::vector<AddressExclusion> exclusionVector;
 		state std::set<AddressExclusion> exclusionSet;
-		bool force = false;
+		state std::vector<AddressExclusion> exclusionAddresses;
+		state std::unordered_set<std::string> exclusionLocalities;
+		state std::vector<std::string> noMatchLocalities;
+		state bool force = false;
 		state bool waitForAllExcluded = true;
 		state bool markFailed = false;
+		state std::vector<ProcessData> workers = wait(makeInterruptable(getWorkers(db)));
 		for (auto t = tokens.begin() + 1; t != tokens.end(); ++t) {
 			if (*t == LiteralStringRef("FORCE")) {
 				force = true;
 			} else if (*t == LiteralStringRef("no_wait")) {
 				waitForAllExcluded = false;
 			} else if (*t == LiteralStringRef("failed")) {
 				markFailed = true;
+			} else if (t->startsWith(ExcludeLocalityKeyDcIdPrefix) ||
+			           t->startsWith(ExcludeLocalityKeyMachineIdPrefix) ||
+			           t->startsWith(ExcludeLocalityKeyProcessIdPrefix) ||
+			           t->startsWith(ExcludeLocalityKeyZoneIdPrefix)) {
+				std::set<AddressExclusion> localityAddresses = getAddressesByLocality(workers, t->toString());
+				if (localityAddresses.empty()) {
+					noMatchLocalities.push_back(t->toString());
+				} else {
+					// add all the server ipaddresses that belong to the given localities to the exclusionSet.
+					std::copy(localityAddresses.begin(), localityAddresses.end(), back_inserter(exclusionVector));
```

Comment:
nit: this is slightly more efficient:

```
exclusionVector.insert(exclusionVector.end(), localityAddresses.begin(), localityAddresses.end());
```

It's more efficient because `std::vector` can pre-allocate with one call to reserve in a way that `std::copy` cannot. I don't think it would help in this particular case, but there are also compilation time wins when you can avoid including `<algorithm>` because it is a giant header.